.github/workflows/run_tests_ubuntu.yml

```diff
@@ -888,3 +888,56 @@ jobs:
           cd build
           LD_LIBRARY_PATH=${LD_LIBRARY_PATH} ctest -j 12 --rerun-failed --output-on-failure -VV
         if: ${{ failure() }}
+
+  fetch_content_cmake:
+    needs: build-deps-serial
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        hdf5: [ 1.14.3 ]
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Install System dependencies
+        shell: bash -l {0}
+        run: sudo apt update && sudo apt install -y libaec-dev zlib1g-dev automake autoconf libcurl4-openssl-dev libjpeg-dev wget curl bzip2 m4 flex bison cmake libzip-dev mpich libmpich-dev 
+
+      ###
+      # Set Environmental Variables
+      ###
+
+      - run: echo "CMAKE_PREFIX_PATH=${HOME}/environments/${{ matrix.hdf5 }}/" >> $GITHUB_ENV
+      - run: echo "LD_LIBRARY_PATH=${HOME}/environments/${{ matrix.hdf5 }}/lib" >> $GITHUB_ENV
+
+      ###
+      # Fetch Cache
+      ###
+
+      - name: Fetch HDF Cache
+        id: cache-hdf5
+        uses: actions/cache@v3
+        with:
+          path: ~/environments/${{ matrix.hdf5 }}
+          key: hdf5-${{ runner.os }}-${{ matrix.hdf5 }}
+
+      - name: Check Cache
+        shell: bash -l {0}
+        run: ls ${HOME}/environments/${{ matrix.hdf5 }} && ls ${HOME}/environments/${{ matrix.hdf5}}/lib
+
+      ###
+      # Configure and build
+      ###
+      - name: Run Cmake 
+        run: |
+          cd integration_tests/fetch_content
+          cmake -S . -B build
+
+      - name: Build
+        run: |
+          cd integration_tests/fetch_content
+          cmake --build build --parallel
```

Comment:
Preferably these tests are run as part of ctest using `ctest --build-and-test` with a `FETCHCONTENT_SOURCE_DIR_<uppercaseName>` option. That way you are checking the current version of the project is  working with fetchcontent.

For an example: https://github.com/LecrisUT/CMake-Template/blob/a4f73dbd53d8cb98e6325296c2905b2ed40dcaa3/test/CMakeLists.txt#L114-L131

Reply:
Oh neat! I'll give that a try tomorrow!

Reply:
@LecrisUT I actually don't really understand what that test is doing and how it works. The stuff I added builds a small example project. From what I understand of what you sent, it seems like it would only build the netcdf project but not actually link to and use it. I feel like the integration test is a complete test. What am I missing here?

Reply:
The first link there is the general framework for an arbitrary (regression) test that runs a cmake project as if it's used by a user from tge ground up. In the template it's ueed for packaging, example and regular regression test.

For an example of a packaging test, see [this CMakeLists.txt](https://github.com/LecrisUT/CMake-Template/blob/a4f73dbd53d8cb98e6325296c2905b2ed40dcaa3/test/package/FetchContent/CMakeLists.txt). There is a bit of fluff to allow testing in the build tree and outside, but if you look closely it's very minimal, FetchContent + link libraries + simple smoke test. See the parent folder for how the `CMakeLists` project is registered as a test.

In the framework link I've sent before, notice the `--test-command ${CMAKE_CTEST_COMMAND}`. That instructs that the test does configure -> build -> ctest. That way you xan be as thorough as you want, e.g. checking that project version matches library, checking features are enabled in the library, etc. This gives more navigable tests since it's just a ctest output run from a ctest test.

Compared to your approach, it's more or less identical, but this adds the ability to be called within the ctest itself, which will make it much easier to keep track of active cmake changes that break the test (it uses the current source as the "git commit" to FetchContent). Your current approach only tests `main` at any given point.

Reply:
No worries! I like detailed explanations and yours makes sense to me. Thanks for the input!